examples/src/main/python/avro_inputformat.py

```diff
@@ -75,7 +75,7 @@
         schema_rdd = sc.textFile(sys.argv[2], 1).collect()
         conf = {"avro.schema.input.key": reduce(lambda x, y: x + y, schema_rdd)}
 
-    avro_rdd = sc.newAPIHadoopFile(
+    avro_rdd = sc.newAPIHadoopFile(  # type: ignore[var-annotated]
```

Comment:
It's good that mypy tests can pass on examples but I wonder if we can remove such `ignore`. Examples are supposed to be verbose and close to the standard. cc @holdenk FYI in case you have some opinions on this.

Reply:
I don't know how the python annotations related to underlying types - would love to learn! can we change the code to make this not a warning otherwise?

Reply:
The code is generic, in a sense that we don't know what `avro_rdd` actually is, other than being `RDD[Tuple[Any, Any]]`. Since we depend on passing strings and reading schema from file, there is really no way to fill `Any` here, which makes the whole annotations useless in practice. But 


```python
from typing import Any, Tuple
...
from pyspark.rdd import RDD
...
avro_rdd: RDD[Tuple[Any, Any]] = ...
```
should silence this one without ignore.

The real question is probably what we mean by 

> verbose and close to the standard.

and what style of coding we want to promote. 

There is a lot of value in annotating library code. However, when it comes to simple scripts like our examples, I wouldn't really expect users to add annotations, maybe with exception to annotation functions.

Also, there is another consideration here ‒ at least some of these are included in the examples. If we decide to prefer annotated code instead of ignores (or annotate things in general), all required type imports will have to go into rendered examples, to make things copy-pasteable. It is not only quite verbose and adds a lot of noise to otherwise simple snippets, but also is related to the problems that we discussed offline.

Finally, some ignores are probably unavoidable, unless we want to ban certain methods (talking about you, `DataFrame.head`) from examples whatsoever,  because of the API design.

Reply:
How about just not running checks on examples if they're to be treated differently? Then there aren't even ignore comments in them

Reply:
> How about just not running checks on examples if they're to be treated differently?

That's an option, but one I'd personally like to avoid. That's because examples that are most likely to cause problems, are the ones corresponding to the "old" API's, which are the trickiest annotate (these depend heavily on generics, including generics `self`, interacting with user code) and have the least comprehensive coverage.

Just to stress this point ‒ my main motivation to add this tests is not to validate `examples` (this are stable, highly exposed and majority of them has been around for long enough to assert that they're OK), but ensure that annotations are consistent with common usage patterns. 

I definitely won't fight over that, but I am really not convinced that targeted ignores are a problem, especially when we have tons of these (and a huge bag of `casts`) within pyspark code.

Reply:
Seems OK to me